scripts/pylib/twister/harness.py

```diff
@@ -121,6 +133,63 @@ def handle(self, line):
 
         self.process_test(line)
 
+class Pytest(Harness):
+
+    def handle(self, line):
+        ''' To keep artifacts of pytest in self.running_dir, pass this directory
+            by "--cmdopt". On pytest end, add a command line option and provide
+            the cmdopt through a fixture function
+			pytest handle output as a whole, it's not necessary to handle it
+			line by line
+        '''
+        global pytest_running_dir
+        if self.running_dir is pytest_running_dir:
+            return
+
+        pytest_running_dir = self.running_dir
+        cmd = [
+			'pytest',
+			'-s',
+			os.path.join(self.source_dir, self.pytest_root),
+			'--cmdopt',
+			self.running_dir,
+			'--junit-xml',
+			os.path.join(self.running_dir, 'report.html'),
+			'-q'
+        ]
+
+        with subprocess.Popen(cmd) as proc:
```

Comment:
this will generate lots of noise
```
i9:pytest_sample((HEAD detached from upstream/pull/29427)): twister -p qemu_x86 -T .
INFO    - Zephyr version: zephyr-v2.5.0-838-gba4a8fc61df2
INFO    - JOBS: 20
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 1 test scenarios (1 configurations) selected, 0 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
handle cmdopt:
/home/nashif/Work/zephyrproject/zephyr/tests/pytest/pytest_sample/twister-out/qemu_x86/tests/pytest/pytest_sample/pytest.sample
run test cases in:
/home/nashif/Work/zephyrproject/zephyr/tests/pytest/pytest_sample/twister-out/qemu_x86/tests/pytest/pytest_sample/pytest.sample
.
- generated xml file: /home/nashif/Work/zephyrproject/zephyr/tests/pytest/pytest_sample/twister-out/qemu_x86/tests/pytest/pytest_sample/pytest.sample/report.html -
1 passed in 0.01s
INFO    - Total complete:    1/   1  100%  skipped:    0, failed:    0
INFO    - 1 of 1 test configurations passed (100.00%), 0 failed, 0 skipped with 0 warnings in 19.91 seconds
INFO    - In total 1 test cases were executed, 0 skipped on 1 out of total 330 platforms (0.30%)
INFO    - 1 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing xunit report /home/nashif/Work/zephyrproject/zephyr/tests/pytest/pytest_sample/twister-out/twister.xml...
INFO    - Writing xunit report /home/nashif/Work/zephyrproject/zephyr/tests/pytest/pytest_sample/twister-out/twister_report.xml...
INFO    - Run completed

```

We do not want to see pytest intermediate results in the twister output.

Reply:
fixed in #33227